### PR TITLE
Fix/sanitize form

### PR DESCRIPTION
**Pull request by carlosdimatteo**
* sanitize input at the element level when its type number
* added disabled states to buttons by validations on anvil components 
* removed unecessary clicking animation on non clickable element 
* some comments

### CODE REVIEW

File: src/components/backstop/BackstopDepositAnvil.tsx
Comment (mootz12) [inline src/components/backstop/BackstopDepositAnvil.tsx:92]: not sure I am a massive fan of the logic being placed here for all anvils

I think it will end up getting moved up once we attribute tooltips to the disabled states, so I'm not gonna block this PR for the code style change.

File: src/components/common/InputBar.tsx
Comment (mootz12) [inline src/components/common/InputBar.tsx:33]: can we drop the log?

File: src/components/lend/LendAnvil.tsx
Comment (mootz12) [inline src/components/lend/LendAnvil.tsx:64]: which number? the newEffectiveCollateral?

File: src/hooks/index.tsx
Comment (mootz12) [inline src/hooks/index.tsx:3]: is there a more common way to use LS?

This was just a quick way I saw to generate functions that allowed functions to persist state to LS.

I don't think we have any uses of this where it affects renders, at the moment

File: src/layouts/DefaultLayout.tsx
Comment (mootz12) [inline src/layouts/DefaultLayout.tsx:25]: fair point, feel free to remove the async wrapper function

File: src/layouts/DefaultLayout.tsx
Comment (mootz12) [inline src/layouts/DefaultLayout.tsx:37]: this chunk sets the `lastPoolId`.

That is, if you route to a page via the nav menu, we will assume the poolId is the last pool you routed too, or the first pool in the reward zone.

**Review by mootz12 — CHANGES_REQUESTED** [review]
LG2M - I think we should strip the extra log before merging.

If you want to pull the disabled logic up, feel free, otherwise whomever implements the tooltips can make their own decision on the best way to organize that.

File: src/components/lend/LendAnvil.tsx
Comment (carlosdimatteo) [reply src/components/lend/LendAnvil.tsx:64]: yes , the values that are only calculated in that moment when the value changes but the conditions for calculating it are not met for example

File: src/hooks/index.tsx
Comment (carlosdimatteo) [reply src/hooks/index.tsx:3]: doing a hook to access LS is fine, what might be overhead is having the values both on state and LS , if put to state its usually to generate reactivity

File: src/components/backstop/BackstopDepositAnvil.tsx
Comment (carlosdimatteo) [reply src/components/backstop/BackstopDepositAnvil.tsx:92]: moved it

File: src/layouts/DefaultLayout.tsx
Comment (carlosdimatteo) [reply src/layouts/DefaultLayout.tsx:37]: yes the thing is this lastPool state is not read anywhere

Comment (mootz12) [reply src/layouts/DefaultLayout.tsx:32]: It's read in `NavMenu`

**Review by mootz12 — APPROVED** [review]
Changes LG2M.

I can work on addressing a few of the TODOs as a followup.

**Pull request merged**
